### ISSUE TITLE
TinkerGraph Example under Example Repo Only Produces a Single Node

### ISSUE THREAD

**Issue by jz48** [plume-oss/plume#91]
I'm using exactly the same example from the plume-getstart-basicusage, however, when I try to export the cpg use driver.exportGraph("./graph3.json"), it only include the node META-DATA. 
How should I fix it?
Just run with 'mvn clean compile exec:java'

<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>de.noshak.plume_ex</groupId>
  <artifactId>plume-example-maven</artifactId>
  <version>1.0-SNAPSHOT</version>

  <properties>
    <maven.compiler.source>11</maven.compiler.source>
    <maven.compiler.target>11</maven.compiler.target>
    <exec.mainClass>de.noshak.plume_ex.Tinker</exec.mainClass>
  </properties>

  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.github.plume-oss</groupId>
        <artifactId>plume</artifactId>
        <version>0.0.3</version>
        <exclusions>
          <exclusion>
            <groupId>io.github.plume-oss</groupId>
            <artifactId>cpgconv</artifactId>
          </exclusion>
        </exclusions>
      </dependency>
    </dependencies>
  </dependencyManagement>

  <dependencies>
    <dependency>
      <groupId>io.github.plume-oss</groupId>
      <artifactId>plume</artifactId>
    </dependency>
  </dependencies>

  <repositories>
    <repository>
      <id>jcenter</id>
      <url>https://jcenter.bintray.com/</url>
    </repository>
  </repositories>

</project>

**Comment by DavidBakerEffendi** [comment]
The documentation needs an update since the release of Plume 0.3.0 :laughing: Also, Plume has moved away from Bintray to Jitpack since Bintray announced that hosting on jcenter will no longer be free after a certain point this year. 

Here is an example repository that I have update to the latest version of Plume and have tested to make sure the JSON works: https://github.com/plume-oss/plume-examples. Let me know if you have any issues with this.

**Comment by jz48** [comment]
> The documentation needs an update since the release of Plume 0.3.0 😆 Also, Plume has moved away from Bintray to Jitpack since Bintray announced that hosting on jcenter will no longer be free after a certain point this year.
> 
> Here is an example repository that I have update to the latest version of Plume and have tested to make sure the JSON works: https://github.com/plume-oss/plume-examples. Let me know if you have any issues with this.

I got it, thanks so much for your reply! I would try it soon.

**Comment by jz48** [comment]
> The documentation needs an update since the release of Plume 0.3.0 😆 Also, Plume has moved away from Bintray to Jitpack since Bintray announced that hosting on jcenter will no longer be free after a certain point this year.
> 
> Here is an example repository that I have update to the latest version of Plume and have tested to make sure the JSON works: https://github.com/plume-oss/plume-examples. Let me know if you have any issues with this.

Thanks so much for your help and it works.

Could I ask another question about plume TinkerGraph?
When I generate the cpg, I find that only includes node typr, doesn't have the node content/value.
For instance, here is a statement 'int a = 5'
Parse it into cpg, 'a' node is a variable and have a type 'indentifier' and value 'a'. However, the result now only have the type 'identifier', toghther with some other info.

Is there any way to generate the node value together with the cpg? Or could I do anything to achieve this?

Thanks so much for your help!

**Comment by DavidBakerEffendi** [comment]
So currently Plume is still in the process of migrating to the standard ShiftLeft compliant CPGs. In your case the statement `int a =5` would produce something like:
```
IDENTIFIER(a)-REF->LOCAL(a)
CALL("<operator>.assignment")-AST->IDENTIFIER(a)
                                                   -AST->LITERAL(5)
```
Whether this is correct or not I am in the process of sorting out and making it similar to Ocular/Joern output

**Comment by jz48** [comment]
Hi David, I'm sorry to bother you again, however, I meet the same issue again when I try to check your comment before about the output structure.

For this time, I pull the repo plume-examples and run ./gradlew tinkerGraphApp in the root folder.

However, there is still only one line with label 'META_DATA' generated in xml and 10 lines in json for the Basic1.java, without the real cpg nodes and edges.

What's wrong with that and how should I fix that?

**Comment by DavidBakerEffendi** [comment]
Okay so I've just upgraded Plume version and added the logging dependencies (so now you can set it to debug logging too). I have performed a clean pull and ran `./gradlew tinkerGraphApp` then ran `cat graph.xml` and I see loads of lines etc. 

If you still run into this issue then could you post your logs set to debug? The configuration can be found under `src/main/resources/log4j2.xml`. Set the following:
```xml
        <Logger name="io.github.plume.oss" level="info" additivity="false">
            <AppenderRef ref="console"/>
        </Logger>
```
to
```xml
        <Logger name="io.github.plume.oss" level="debug" additivity="false">
            <AppenderRef ref="console"/>
        </Logger>
```

**Comment by jz48** [comment]
I tried to do so as you suggested and the result is shown in following pictures:

<img width="1680" alt="Screen Shot 2021-03-17 at 10 23 07 PM" src="https://user-images.githubusercontent.com/70977171/111483178-8789b280-876f-11eb-8d7e-aa560e325d75.png">

<img width="1680" alt="Screen Shot 2021-03-17 at 9 54 55 PM" src="https://user-images.githubusercontent.com/70977171/111483696-ff57dd00-876f-11eb-85ec-0f74f1d7d29d.png">


Basically, it told me no class found in the example file.

**Comment by DavidBakerEffendi** [comment]
That is a really weird file name - are you running on an M1 Mac? I know that if a source file is given then it is first compiled, moved to `/tmp/plume/build`, analyzed, and then the temp files are cleaned again. Perhaps the architecture might be responsible for complications. I should also add logging on which files are compiled and loaded.

Plume is ready for another small release so when that is up I'll add better logging to help debug the problem.

**Comment by jz48** [comment]
No, I'm running on MBP-2016 and MacOS Big Sur v11.2.3

Does the source file have to be able to be compiled?

I'm going to try it on windows also.

**Comment by DavidBakerEffendi** [comment]
So Plume only analyzes the bytecode so for convenience, if it is given a source file it will attempt to get the system's JDK to compile it. All the unit tests run Linux

**Comment by DavidBakerEffendi** [comment]
Note that this is sensitive to the file extension

**Comment by DavidBakerEffendi** [comment]
@jz48 Just following up, you manage to figure out the issue?

**Comment by jz48** [comment]
Hi David, I try it on ubuntu and it works. The example does generate the cpg with nodes and edges! There is no doubt that it works.
On the other hand, I find that (maybe) it could only deal with the compilable source file. For others, it would generate an empty cpg case compile fail.
Thanks so much for you patient and kindly answer and reply! Wish you could make it batter and better!

**Comment by DavidBakerEffendi** [comment]
Thank you! I'm happy to hear you got it sorted. I think if anything I should add Windows and OSX to the CI/CD!

**Comment by DavidBakerEffendi** [comment]
@jz48 This PR: https://github.com/plume-oss/plume/pull/126 fixes the issue. Essentially the temporary directories given back by Java did not end in a file separator in Linux but did in Mac and Windows - super weird! But this did highlight a few more issues further down the line anyway but now Plume tests on these two OS's now too. Release v0.3.7 is on the pipeline now and should be released soon to address this!
